### PR TITLE
Run shellcheck on all shell scripts

### PR DESCRIPTION
**Pull request by chizhg**
## Proposed Changes
- Run `shellcheck` on all shell scripts
`shellcheck -f diff test/*.sh | git apply`
`shellcheck -f diff hack/*.sh | git apply`
- The new version of `ko` has a nice support for Go modules, then there is no need to `sed` when uploading eventing test images
- Fixes https://github.com/google/knative-gcp/issues/702

**Release Note**

```release-note
NONE
```

### CODE REVIEW

File: hack/verify-codegen.sh [hunk 1]
```diff
@@ -52,9 +52,8 @@ diff -Naupr --no-dereference \
 diff -Naupr --no-dereference \
   "${REPO_ROOT_DIR}/pkg" "${TMP_DIFFROOT}/pkg" || ret=1
 
-# TODO uncomment this once we fix https://github.com/google/knative-gcp/issues/702
-# diff -Naupr --no-dereference \
-#  "${REPO_ROOT_DIR}/third_party" "${TMP_DIFFROOT}/third_party" || ret=1
+diff -Naupr --no-dereference \
```
Comment (liu-cong) [inline hack/verify-codegen.sh:55]: I see #702 is still open. Is it fixed?

File: hack/lib.sh [hunk 2]
```diff
@@ -153,18 +153,21 @@ function storage_admin_set_up() {
   local project_id=${1}
   local pubsub_service_account=${2}
   local pubsub_service_account_key_temp=${3}
```
Comment (liu-cong) [inline hack/lib.sh:155]: Looks like this param is not needed any more?

File: hack/verify-codegen.sh (hunk 1 above)
Comment (liu-cong) [reply hack/verify-codegen.sh:55]: Sorry I didn't notice this PR actually fixes it

File: hack/lib.sh (hunk 2 above)
Comment (chizhg) [reply hack/lib.sh]: Good catch, removed.

File: hack/verify-codegen.sh (hunk 1 above)
Comment (chizhg) [reply hack/verify-codegen.sh:55]: Yeah it was fixed a while ago.

File: hack/init_cloud_storage_source.sh [hunk 3]
```diff
@@ -21,18 +21,9 @@ set -o errexit
...
 
-# Download a JSON key for the service account.
-gcloud iam service-accounts keys create "${PUBSUB_SERVICE_ACCOUNT_KEY_TEMP}" \
-  --iam-account="${PUBSUB_SERVICE_ACCOUNT}"@"${PROJECT_ID}".iam.gserviceaccount.com
-
-storage_admin_set_up "${PROJECT_ID}" "${PUBSUB_SERVICE_ACCOUNT}" "${PUBSUB_SERVICE_ACCOUNT_KEY_TEMP}"
-
-# Remove the tmp file.
-rm "${PUBSUB_SERVICE_ACCOUNT_KEY_TEMP}"
+storage_admin_set_up "${PROJECT_ID}" "${PUBSUB_SERVICE_ACCOUNT}"
```
Comment (ian-mi) [inline hack/init_cloud_storage_source.sh:29]: Why these changes?

Comment (chizhg) [reply hack/init_cloud_storage_source.sh:29]: [This line](https://github.com/google/knative-gcp/blob/master/hack/lib.sh#L166) is too long and some logic is unnecessary to just get the GCS service account name, so I simplified it a bit. And Cong pointed out that one arg in this function is needed anymore after my change, so I also removed it from here.

May need @capri-xiyue to take a second look.

File: hack/lib.sh [hunk 4]
```diff
@@ -128,43 +128,44 @@ function enable_workload_identity(){
...
     --role roles/iam.serviceAccountAdmin
   }
 
 function storage_admin_set_up() {
   echo "Update ServiceAccount for Storage Admin"
   local project_id=${1}
   local pubsub_service_account=${2}
-  local pubsub_service_account_key_temp=${3}
+  local project_number="$(gcloud projects describe "${project_id}" --format="value(projectNumber)")"
+  local access_token="$(gcloud auth application-default print-access-token)"
```
Comment (capri-xiyue) [inline hack/lib.sh:156]: Which credential does this command use? The credential for the control plane or the credential for the data plane?
https://cloud.google.com/sdk/gcloud/reference/auth/application-default/print-access-token. 
In the docs, it looks like it may not always be the credential for the data plane.

File: hack/lib.sh [hunk 5]
```diff
@@ -128,43 +128,44 @@ function enable_workload_identity(){
...
-  gcloud projects add-iam-policy-binding ${project_id} \
-    --member=serviceAccount:${pubsub_service_account}@${project_id}.iam.gserviceaccount.com \
+  gcloud projects add-iam-policy-binding "${project_id}" \
+    --member=serviceAccount:"${pubsub_service_account}"@"${project_id}".iam.gserviceaccount.com \
     --role roles/storage.admin
-  export GCS_SERVICE_ACCOUNT=`curl -s -X GET -H "Authorization: Bearer \`GOOGLE_APPLICATION_CREDENTIALS=${pubsub_service_account_key_temp} gcloud auth application-default print-access-token\`" "https://www.googleapis.com/storage/v1/projects/${project_id}/serviceAccount" | grep email_address | cut -d '"' -f 4`
-  gcloud projects add-iam-policy-binding ${project_id} \
-    --member=serviceAccount:"${GCS_SERVICE_ACCOUNT}" \
+
+  curl -s -X GET -H "Authorization: Bearer ${access_token}" "https://www.googleapis.com/storage/v1/projects/${project_id}/serviceAccount"
```
Comment (capri-xiyue) [inline hack/lib.sh:167]: Why do we still need this command here?

File: hack/lib.sh [hunk 6]
```diff
@@ -128,43 +128,44 @@ function enable_workload_identity(){
...
+  gcloud projects add-iam-policy-binding "${project_id}" \
```
Comment (capri-xiyue) [inline hack/lib.sh:168]: We need to update the docs accordingly https://github.com/google/knative-gcp/blob/master/docs/examples/cloudstoragesource/README.md

File: hack/init_cloud_storage_source.sh (hunk 3 above)
Comment (capri-xiyue) [reply hack/init_cloud_storage_source.sh:29]: Have some questions regarding the removal of ${PUBSUB_SERVICE_ACCOUNT_KEY_TEMP}. Already added some comments in lib.sh

File: hack/lib.sh (hunk 5 above)
Comment (chizhg) [reply hack/lib.sh]: This command is needed to activate the GCS service account.

File: hack/lib.sh (hunk 6 above)
Comment (chizhg) [reply hack/lib.sh:168]: The doc does not need to be updated. Here it's using Option 1 to get the GCS service account, except instead of visiting the pantheon UI, it's using the curl command to activate the service account.

File: hack/lib.sh (hunk 4 above)
Comment (chizhg) [reply hack/lib.sh]: You are right, I have added the service account key back and used it to get the `access_token`.

**Comment by liu-cong** [comment]
/lgtm

I will defer to @capri-xiyue to approve
